src/core/ddsc/src/dds_qos.c

```diff
@@ -618,8 +618,20 @@ bool dds_qget_partition (const dds_qos_t * __restrict qos, uint32_t *n, char ***
     *n = qos->partition.n;
   if (ps)
   {
-    if (qos->partition.n == 0)
-      *ps = NULL;
+    if (qos->partition.n == 0) {
+      /* This is the default partition.
+       * Only when the argument n is non-null, we'll
+       * actually return the default partition (i.e.,
+       * the empty string "").
+       *  */
+      if (n) {
```

Comment:
Lines 615 & 616 ensure (ps ≠ `NULL`) ⇒ (n ≠ `NULL`), so we are certain to never end up in the `else` case. I think the comment that says "this is the default partition" might better be rephrased as "this implies the default partition", then the remainder of the comment, the `if (n)` and the `else` branch can be thrown out.